docs/L1_L2_Messages.md

```diff
@@ -89,127 +93,41 @@ If the Retryable Ticket is cancelled or expires before it is redeemed, Callvalue
-    * @return (price, nextUpdateTimestamp). Price is guaranteed not to change until nextUpdateTimestamp.
-    */
-    function getSubmissionPrice(uint calldataSize) external view returns (uint, uint);
-
-    /**
-     * @notice Return the price, in wei, of extending the lifetime of txId by an additional lifetime period. Revert if txId doesn't exist.
-     * @param txId  unique identifier of retryabale message: keccak256(requestID, uint(0) )
-     * @return (price, nextUpdateTimestamp). Price is guaranteed not to change until nextUpdateTimestamp.
-    */
-    function getKeepalivePrice(bytes32 txId) external view returns(uint, uint);
+The Arbitrum protocol's usage of L2 Aliases for L1-to-L2 messages prevents cross-chain exploits that would otherwise be possible if we simply reused L1 contact addresses.
 
-    /**
-    @notice Deposits callvalue into the sender's L2 account, then adds one lifetime period to the life of txId.
-    * If successful, emits LifetimeExtended event.
-    * Revert if txId does not exist, or if the timeout of txId is already at least one lifetime in the future, or if the sender has insufficient funds (after the deposit).
-    * @param txId unique identifier of retryabale message: keccak256(requestID, uint(0) )
-    * @return New timeout of txId.
-    */
-    function keepalive(bytes32 txId) external payable returns(uint);
+If for some reason you need to compute an L2 alias's corresponding L1 address on chain, you can use our `AddressAliasHelper` library:
 
-    /**
-    * @notice Return the beneficiary of txId.
-    * Revert if txId doesn't exist.
-    * @param txId unique identifier of retryabale message: keccak256(requestID, uint(0) )
-    * @return address of beneficiary for transaction
-    */
-    function getBeneficiary(bytes32 txId) external view returns (address);
-
-    /**
-    @notice Cancel txId and refund its callvalue to its beneficiary.
-    * Revert if txId doesn't exist, or if called by anyone other than txId's beneficiary.
-    @param txId unique identifier of retryabale message: keccak256(requestID, uint(0) )
-    */
-    function cancel(bytes32 txId) external;
+```sol
+    modifier onlyFromMyL1Contract() override {
+        require(AddressAliasHelper.undoL1ToL2Alias(msg.sender) === myL1ContractAddress, "ONLY_COUNTERPART_CONTRACT");
+        address _inbox = inbox;
+        _;
+    }
+```
 
-    event LifetimeExtended(bytes32 indexed txId, uint newTimeout);
-    event Redeemed(bytes32 indexed txId);
-    event Canceled(bytes32 indexed txId);
-}
+_Of note: the (highly) recommended convenience methods `Inbox.depositEth` and `Inbox.createRetryableTicket` handle subtracting the offset for the credit back address and beneficiary address, and using the these reverse-aliased values as the inputs for the create retryable tickets. This way, the beneficiary and credit back addresses as received on L2 will be equal to those provided as the L1 params._
```

Comment:
I'm not sure "will be equal" is right (as they are different addresses).
will be equivalent?